src/Storages/StorageSQLite.cpp

```diff
@@ -159,9 +166,9 @@ void registerStorageSQLite(StorageFactory & factory)
         const auto database_path = engine_args[0]->as<ASTLiteral &>().value.safeGet<String>();
         const auto table_name = engine_args[1]->as<ASTLiteral &>().value.safeGet<String>();
 
-        auto sqlite_db = openSQLiteDB(database_path, args.getContext());
+        auto sqlite_db = openSQLiteDB(database_path, args.getContext(), /* throw_on_error */!args.attach);
```

Comment:
If we open in `StorageSQLite`, maybe we can move call of `openSQLiteDB` from here?

Reply:
No, because it is needed not only in StorageSQLite. Database engine also uses it separately.